### ISSUE TITLE
PGN calculation  (DP + PF + PS) or (R + DP + PF + PS)?

### ISSUE THREAD

**Issue by aldas** [canboat/canboat#248]
SAE J1939 / NMEA? spec says that  PGN is calculated as `R + DP + PF + PS` but for PGN calculation Canboat  does not seems to use `R` bit.  https://github.com/canboat/canboat/blob/769160d189d5c1e43dae5608bba2a7c9bb1e083b/common/common.c#L532-L536I understand this is because of documentation seems to say that R (reserved bit) should be always 0 for NMEA message.

maybe for analyser debugging purposes it would be better to include it?

```c
unsigned char RDP = (unsigned char) (id >> 24) & 3;  // use R + DP bits 
``` 

I noticed this because I tried to decode ZF throttle lever can/j1939 messages (yes I know now it is not a nmea2k device) and could not understand why different tools gives me different PGNs.

**Comment by arks-public** [comment]
I agree that the PGN should be calculated as (R + DP + PF + PS). For NMEA 2000, R should always be zero, so it wouldn't affect the interpretation of NMEA 2000 messages as long as they are formatted correctly. Plus, it seems like you would want to know if a NMEA 2000 device is incorrectly setting R=1, rather than ignoring that bit. The underlying standards such as J1939 do refer to that bit as Extended Data Page (EDP) instead of Reserved (R), for example see [here](https://assets.vector.com/cms/content/know-how/_application-notes/AN-ION-1-3100_Introduction_to_J1939.pdf) pages 3-4 or [here](https://www.sis.se/api/document/preview/80007651/) last 2 pages.
